src/raftstore/store/worker/compact.rs

```diff
@@ -11,40 +11,43 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use raftstore::store::{PeerStorage, keys};
+use raftstore::store::keys;
 use raftstore::store::engine::Iterable;
 use util::worker::Runnable;
 use util::rocksdb;
-use storage::CF_RAFT;
+use storage::{CF_RAFT, CF_LOCK};
 
 use rocksdb::{DB, WriteBatch, Writable};
 use std::sync::Arc;
 use std::fmt::{self, Formatter, Display};
 use std::error;
 
-/// Compact task.
-pub struct Task {
-    engine: Arc<DB>,
-    region_id: u64,
-    compact_idx: u64,
-}
-
-impl Task {
-    pub fn new(ps: &PeerStorage, compact_idx: u64) -> Task {
-        Task {
-            engine: ps.get_engine().clone(),
-            region_id: ps.get_region_id(),
-            compact_idx: compact_idx,
-        }
-    }
+pub enum Task {
+    CompactLockCF {
+        engine: Arc<DB>,
```

Comment:
I think we can create the engine field in Runner, no need to pass it every time in Task.